### PR TITLE
Self Harm Toggle

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

Similarly to another PR made by a maintainer (The same maintainer that helped me make this one.), this PR prevents you from accidentally hitting yourself. Except this one, like, actually works.

It adds a keybind for toggling self-harm. It prevents you from hitting yourself while on harm intent while unarmed and hitting yourself at all while holding a weapon. Eating, healing, or hitting yourself with anything else still works just fine. (And if you want to do suicide, simply toggle it back on.) It defaults to on.

## Why It's Good For The Game

Maybe because hitting yourself with a harvester mid-fight and breaking your bones isn't exactly, well, great? (Or any weapon, for that matter.)

## Changelog
:cl:
add: new keybind for toggling self-harm, defaults to on
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/weapon.dm [hunk 1]
```diff
@@ -10,3 +10,9 @@
 	var/max_shots = 0 //codex, energy weapons
 	var/scope_zoom = FALSE//codex
 	var/self_recharge = FALSE //codex
+
+/obj/item/weapon/melee_attack_chain(mob/user, atom/target, params, rightclick)
+	if(target == user && !user.do_self_harm)
+		return
+	else
+		return ..()
```
Comment (BraveMole) [inline code/game/objects/items/weapons/weapon.dm:18]:
```suggestion
	return ..()
```
the else is useless in that context

File: code/modules/mob/mob_defines.dm [hunk 2]
```diff
@@ -93,3 +93,6 @@
 	var/has_unlimited_silicon_privilege = 0
 	///The faction this mob belongs to
 	var/faction = FACTION_NEUTRAL
+
+	// Whether or not the mob can hit themselves.
+	var/do_self_harm = TRUE
```
Comment (BraveMole) [inline code/modules/mob/mob_defines.dm:98]: should probably put that var near in_throw_mode

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by RikuTheKiller** [comment]
Guess I'll just fetch upstream while I'm here.

**Comment by Texicos** [comment]
they removed the fucking skill issue from TGMC, can't have shit

**Pull request merged**